src/torchmetrics/utilities/data.py

```diff
@@ -174,7 +174,7 @@ def _bincount(x: Tensor, minlength: Optional[int] = None) -> Tensor:
         - deterministic mode on GPU.
         - MPS devices
 
-    This implementation fallback to a for-loop counting occurrences in that case.
+    The implementation uses torch.arange for deterministic counting only.
```

Comment:
The word "only" at the end is problematic I think, because the code says otherwise. It also uses the slower implementation in case of MPS and XLA